ocean_lib/models/bpool.py

```diff
@@ -596,29 +597,23 @@ def get_liquidity_logs(
         """
         topic0 = self.get_event_signature(event_name)
         to_block = to_block or "latest"
-        _filter = {"fromBlock": from_block, "toBlock": to_block, "topics": [topic0]}
-        if this_pool_only:
-            _filter["address"] = self.address
+        topics = [topic0]
+        address = self.address if this_pool_only else None
 
         if user_address:
             assert Web3.isChecksumAddress(user_address)
-            _filter["topics"].append(
+            topics.append(
                 f"0x000000000000000000000000{remove_0x_prefix(user_address).lower()}"
             )
-
         event = getattr(self.events, event_name)
-        event_abi = event().abi
-        try:
-            logs = self.web3.eth.get_logs(_filter)
-            logs = [get_event_data(self.web3.codec, event_abi, lg) for lg in logs]
-        except ValueError as e:
-            logger.error(
-                f"get_join_logs failed -> web3.eth.get_logs (filter={_filter}) failed: "
-                f"{e}.."
-            )
-            logs = []
-
-        return logs
+        event_filter = EventFilter(
```

Comment:
The original logic is more similar to `getLogs` than to the `Filter` API.  Plus the `EventFilter` is never reused for polling. Please use `event.getLogs()` instead of `EventFilter`

Reply:
Should we consider removing `this_pool_only` parameter since `address` parameter is not used anymore if `getLogs` don't ask it? As well as `from_all_tokens` from `data_token.py` in `get_start_order_logs` function?

Reply:
Any thoughts here @DMats?

Reply:
For now, we should keep both `this_pool_only` and `from_all_tokens`.  I'll leave another comment on the PR later explaining why in more detail.

Reply:
Kept them already 👍🏻